android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/activity/ActivitySchemaCache.java

```diff
@@ -0,0 +1,156 @@
+    @NonNull
+    private final Map<String, SchemaReference> taskIdentifierToSchema;
+    @NonNull
+    private final Map<String, SurveyReference> surveyIdentifierToSurvey;
+    @NonNull
+    private final Map<String, String> surveyGuidToIdentifier;
+    @NonNull
+    private final Map<String, CompoundActivity> taskIdentifierToCompoundActivity;
+    @NonNull
+    private final Map<String, ActivityType> identifierToActivityType;
+
+
+    public ActivitySchemaCache() {
+        taskIdentifierToSchema = Maps.newHashMap();
+        surveyIdentifierToSurvey = Maps.newHashMap();
+        surveyGuidToIdentifier = Maps.newHashMap();
+        taskIdentifierToCompoundActivity = Maps.newHashMap();
+        identifierToActivityType = Maps.newHashMap();
+
+        addTasks();
+    }
+
+    private void addTasks() {
+        // TODO: these should be configurable
+        identifierToActivityType.put("1-APHMedicationTracker-20EF8ED2-E461-4C20-9024-F43FCAAAF4C3", ActivityType.TASK);
+        taskIdentifierToSchema.put("1-APHMedicationTracker-20EF8ED2-E461-4C20-9024-F43FCAAAF4C3",
+                new SchemaReference().id("Medication Tracker").revision(8L));
+
+        identifierToActivityType.put("1-APHTremor-108E189F-4B5B-48DC-BFD7-FA6796EEf439", ActivityType.TASK);
+        taskIdentifierToSchema.put("1-APHTremor-108E189F-4B5B-48DC-BFD7-FA6796EEf439",
+                new SchemaReference().id("Tremor Activity").revision(3L));
+
+        identifierToActivityType.put("2-APHIntervalTapping-7259AC18-D711-47A6-ADBD-6CFCECDED1DF", ActivityType.TASK);
+        taskIdentifierToSchema.put("2-APHIntervalTapping-7259AC18-D711-47A6-ADBD-6CFCECDED1DF",
+                new SchemaReference().id("Tapping Activity").revision(11L));
+
+        identifierToActivityType.put("3-APHMoodSurvey-7259AC18-D711-47A6-ADBD-6CFCECDED1DF", ActivityType.TASK);
+        taskIdentifierToSchema.put("3-APHMoodSurvey-7259AC18-D711-47A6-ADBD-6CFCECDED1DF",
+                new SchemaReference().id("Mood Survey").revision(2L));
+
+        identifierToActivityType.put("3-APHPhonation-C614A231-A7B7-4173-BDC8-098309354292", ActivityType.TASK);
+        taskIdentifierToSchema.put("3-APHPhonation-C614A231-A7B7-4173-BDC8-098309354292",
+                new SchemaReference().id("Voice Activity").revision(4L));
+
+        identifierToActivityType.put("4-APHTimedWalking-80F09109-265A-49C6-9C5D-765E49AAF5D9", ActivityType.TASK);
+        taskIdentifierToSchema.put("4-APHTimedWalking-80F09109-265A-49C6-9C5D-765E49AAF5D9",
+                new SchemaReference().id("Walking Activity").revision(6L));
+    }
+
+    /**
+     * Caches the activity's schema information according to its type.
+     *
+     * @param activity activity to cache
+     */
+    public void cacheActivitySchema(Activity activity) {
+        switch (activity.getActivityType()) {
+            case TASK:
+                TaskReference taskReference = activity.getTask();
+                if (taskReference != null) {
```

Comment:
You should log an error of taskReference is null here.  Along with the other checks in this switch.